src/main/resources/db/changelog/changes/v1.0.2/fix_user_details_preferred_contact_type.xml

```diff
@@ -0,0 +1,98 @@
+<databaseChangeLog xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+  <changeSet id="create-view-user-details" author="kjain@ebsco.com">
+    <!-- ensure other schemas/tables exist (primarily to prevent invalid references in integration tests) -->
+    <preConditions onFail="CONTINUE">
+      <viewExists viewName="src_users_groups"/>
+      <viewExists viewName="src_users_users"/>
+    </preConditions>
+    <createView
+      replaceIfExists="true"
+      viewName="drv_user_details">
+      SELECT
+      userDetails.jsonb -> 'personal' ->> 'firstName' as user_first_name,
+      userDetails.jsonb -> 'personal' ->> 'lastName' as user_last_name,
+      userDetails.jsonb ->>'barcode' as user_barcode,
+      userDetails.jsonb ->>'username' as username,
+      userDetails.id as id,
+      userDetails.jsonb ->> 'externalSystemId' as user_external_system_id,
+      userDetails.jsonb ->> 'active' as user_active,
+      userDetails.jsonb -> 'personal' ->> 'email' as user_email,
+      userDetails.jsonb -> 'metadata' ->> 'createdDate' as user_created_date,
+      userDetails.jsonb -> 'metadata' ->> 'updatedDate' as user_updated_date,
+      userDetails.jsonb -> 'personal' ->> 'preferredFirstName' as user_preferred_first_name,
+      userDetails.jsonb -> 'personal' ->> 'middleName' as user_middle_name,
+      userDetails.jsonb -> 'personal' ->> 'phone' as user_phone,
+      userDetails.jsonb -> 'personal' ->> 'mobilePhone' as user_mobile_phone,
+      userDetails.jsonb -> 'personal' ->> 'dateOfBirth' as user_date_of_birth,
+      userDetails.jsonb ->> 'expirationDate'::text AS user_expiration_date,
+      userDetails.jsonb ->> 'enrollmentDate'::text AS user_enrollment_date,
+      patron_id_ref_data.jsonb ->> 'group'::text AS user_patron_group,
+      patron_id_ref_data.id::text AS user_patron_group_id,
+      UserDetails.jsonb -> 'personal' ->> 'preferredContactTypeId' as user_preferred_contact_type_id,
+      CASE UserDetails.jsonb -> 'personal' ->> 'preferredContactTypeId'
+      WHEN '001' THEN 'Mail (Primary Address)'
```

Comment:
Earlier we were having 5 preferred contact type but in order to make it consistent along with the user module, we are aligning it these 3 preferred contact type.